tests/tensorflow/test_tensorflow_autolog.py

```diff
@@ -0,0 +1,213 @@
+    assert get_tf_core_random_tensors_run_instance.data.metrics['b'] == 4.0
+
+
+@pytest.fixture
+def tf_estimator_random_data():
+    mlflow.tensorflow.autolog(metrics_every_n_steps=1)
+    run = mlflow.start_run()
+    dir = tempfile.mkdtemp()
+    CSV_COLUMN_NAMES = ['SepalLength', 'SepalWidth', 'PetalLength', 'PetalWidth', 'Species']
+    SPECIES = ['Setosa', 'Versicolor', 'Virginica']
+
+    train_path = tf.keras.utils.get_file(
+        "iris_training.csv", "https://storage.googleapis.com/download.tensorflow.org/data/iris_training.csv")
+    test_path = tf.keras.utils.get_file(
+        "iris_test.csv", "https://storage.googleapis.com/download.tensorflow.org/data/iris_test.csv")
+
+    train = pd.read_csv(train_path, names=CSV_COLUMN_NAMES, header=0)
+    test = pd.read_csv(test_path, names=CSV_COLUMN_NAMES, header=0)
+
+    train_y = train.pop('Species')
+    test_y = test.pop('Species')
+
+    def input_fn(features, labels, training=True, batch_size=256):
+        """An input function for training or evaluating"""
+        # Convert the inputs to a Dataset.
+        dataset = tf.data.Dataset.from_tensor_slices((dict(features), labels))
+
+        # Shuffle and repeat if you are in training mode.
+        if training:
+            dataset = dataset.shuffle(1000).repeat()
+
+        return dataset.batch(batch_size)
+
+    my_feature_columns = []
+    for key in train.keys():
+        my_feature_columns.append(tf.feature_column.numeric_column(key=key))
+
+    feature_spec = {}
+    for feature in CSV_COLUMN_NAMES:
+        feature_spec[feature] = tf.placeholder(dtype="float", name=feature, shape=[150])
+
+    receiver_fn = tf.estimator.export.build_raw_serving_input_receiver_fn(feature_spec)
+
+    classifier = tf.estimator.DNNClassifier(
+        feature_columns=my_feature_columns,
+        # Two hidden layers of 10 nodes each.
+        hidden_units=[30, 10],
+        # The model must choose between 3 classes.
+        n_classes=3,
+        model_dir=dir)
+
+    classifier.train(
+        input_fn=lambda: input_fn(train, train_y, training=True),
+        steps=5000)
+    classifier.export_saved_model(dir, receiver_fn)
+    mlflow.end_run()
+    shutil.rmtree(dir)
+    return run.info.run_id
+
+
+@pytest.fixture
+def get_tf_estimator_random_data_run_instance(tf_estimator_random_data):
+    client = mlflow.tracking.MlflowClient()
+    run = client.get_run(tf_estimator_random_data)
+    return run
+
+
+@pytest.mark.small
+def test_tf_estimator_autolog_logs_metrics(get_tf_estimator_random_data_run_instance):
+    metrics = get_tf_estimator_random_data_run_instance.data.metrics
+    assert len(metrics) > 0
+
+@pytest.mark.large
+def test_tf_estimator_autolog_saves_model(get_tf_estimator_random_data_run_instance):
+    client = mlflow.tracking.MlflowClient()
+    artifacts = client.list_artifacts(get_tf_estimator_random_data_run_instance.info.run_id)
+    artifacts = map(lambda x: x.path, artifacts)
+    assert 'model' in artifacts
```

Comment:
Same comment as for Keras, maybe we can try to load the model back & make predictions, to verify that we logged a valid model?